### PR TITLE
Adding OOOH Link support (#5)

### PR DESCRIPTION
**Pull request by abhinab22mohanty**
* Adding OOOH external link support

1. Added link display support to OutOfOffice Pane
2.  Added flag `openLinkInNewTab`  to enable link opening in new tab if set to true and link opening in widget iframe if set to false

### CODE REVIEW

File: chat-components/src/components/outofofficehourspane/common/presetFourProps/presetFourOOOHPaneControlProps.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { IOOOHPaneControlProps } from "../../interfaces/IOOOHPaneControlProps";
+import { defaultOOOHPaneControlProps } from "../defaultProps/defaultOOOHPaneControlProps";
+
+export const presetFourOOOHPaneControlProps: IOOOHPaneControlProps = {
+    ...defaultOOOHPaneControlProps,
+    id: "oc-lcw-outofofficehourspane-preset1",
+    titleText: "We are out of office. Please contact us here https://www.microsoft.com for further assisstance",
```
Comment (elopezanaya) [inline chat-components/src/components/outofofficehourspane/common/presetFourProps/presetFourOOOHPaneControlProps.ts:7]: have we considerate to localize default messages?

**Review by elopezanaya — APPROVED** [review]

File: chat-components/src/common/utils.ts [hunk 2]
```diff
@@ -95,4 +95,19 @@ export const addNoreferrerNoopenerTag = (htmlNode: any) => {
...
         }
     }
+};
+
+export const replaceURLWithAnchor = (text: string | undefined, openInNewTab: boolean | undefined) => {
+    if (text) {
+        const modifiedText = text.replace(Regex.URLRegex, function(url) {
+            if (openInNewTab) {
+                // eslint-disable-next-line quotes
+                return '<a href="' + url + '" rel="noreferrer noopener" target="_blank">' + url + '</a>';
```
[the remark this answers is not in the document]
Comment (abhinab22mohanty) [reply chat-components/src/common/utils.ts:105]: We are currently using the same Regex for URL that we use for validating URL. https://github.com/microsoft/omnichannel-chat-widget/blob/5d6a45b4f2c53e3f5ab9821294d66b3e6bc1f3e9/chat-components/src/common/utils.ts#L53
So as a part of replace we check for valid URl before replacing or else the text is rendered as is

File: chat-components/src/components/outofofficehourspane/common/presetFourProps/presetFourOOOHPaneControlProps.ts (hunk 1 above)
Comment (abhinab22mohanty) [reply chat-components/src/components/outofofficehourspane/common/presetFourProps/presetFourOOOHPaneControlProps.ts:7]: Unsure on this as this is just a dummy story for VRT displaying and validating this capability

File: chat-components/src/components/outofofficehourspane/common/defaultProps/defaultOOOHPaneControlProps.ts [hunk 3]
```diff
@@ -6,4 +6,5 @@ export const defaultOOOHPaneControlProps: IOOOHPaneControlProps = {
     hideOOOHPane: false,
     hideTitle: false,
     titleText: "Thanks for contacting us. You have reached us outside of our operating hours. An agent will respond when we open.",
+    openLinkInNewTab: false
```
Comment (charliewang95) [inline chat-components/src/components/outofofficehourspane/common/defaultProps/defaultOOOHPaneControlProps.ts:9]: Please update defaultDummyProps.ts for this change

File: chat-components/src/components/outofofficehourspane/OOOHPane.tsx [hunk 4]
```diff
@@ -29,16 +33,14 @@ function OOOHPane(props: IOOOHPaneProps) {
...
-                        <Label
+                        <Text
                             className={props.styleProps?.classNames?.titleClassName}
                             styles={titleStyles}
                             tabIndex={-1}
                             id={elementId + "-title"}>
-                            {props.controlProps?.titleText ?? defaultOOOHPaneControlProps.titleText}
-                        </Label>)}
-
+                            <div dangerouslySetInnerHTML={{ __html: displayText }}></div>
```
Comment (charliewang95) [inline chat-components/src/components/outofofficehourspane/OOOHPane.tsx:42]: Please test accessibility with this change - whether tis is being announced correctly

File: chat-components/src/components/outofofficehourspane/common/defaultProps/defaultOOOHPaneControlProps.ts (hunk 3 above)
Comment (abhinab22mohanty) [reply chat-components/src/components/outofofficehourspane/common/defaultProps/defaultOOOHPaneControlProps.ts:9]: Yes will update once chat-component changes go through will pull latest build with chat-widget and add this prop

File: chat-components/src/components/outofofficehourspane/OOOHPane.tsx (hunk 4 above)
Comment (abhinab22mohanty) [reply chat-components/src/components/outofofficehourspane/OOOHPane.tsx:42]: Verified complete link url is being announced correctly with screen reader

**Review by charliewang95 — APPROVED** [review]

**Pull request merged**